python/cudf/cudf/dataframe/series.py

```diff
@@ -464,62 +461,56 @@ def tail(self, n=5):
 
         return self.iloc[-n:]
 
-    def to_string(self, nrows=NOTSET):
+    def to_string(self):
         """Convert to string
 
-        Parameters
-        ----------
-        nrows : int
-            Maximum number of rows to show.
-            If it is None, all rows are shown.
+        Uses Pandas formatting internals to produce output identical to Pandas.
+        Use the Pandas formatting settings directly in Pandas to control cuDF
+        output.
         """
-        if nrows is NOTSET:
-            nrows = settings.formatting.get(nrows)
+        return self.__repr__()
 
-        str_dtype = self.dtype
+    def __str__(self):
+        return self.to_string()
 
-        if len(self) == 0:
-            return "<empty Series of dtype={}>".format(str_dtype)
+    def __repr__(self):
+        mr = pd.options.display.max_rows
+        if len(self) > mr:
+            top = self.head(int(mr))
+            bottom = self.tail(int(mr))
+            from cudf import concat
 
-        if nrows is None:
-            nrows = len(self)
+            preprocess = concat([top, bottom])
         else:
-            nrows = min(nrows, len(self))  # cap row count
-
-        more_rows = len(self) - nrows
-
-        # Prepare cells
-        cols = OrderedDict([("", self.values_to_string(nrows=nrows))])
-        dtypes = OrderedDict([("", self.dtype)])
-        # Format into a table
-        output = formatting.format(
-            index=self.index,
-            cols=cols,
-            dtypes=dtypes,
-            more_rows=more_rows,
-            series_spacing=True,
-        )
-        if self.name is None:
-            output += "\ndtype: {}".format(str_dtype)
+            preprocess = self
+        if is_categorical_dtype(preprocess.dtype):
+            return preprocess.to_pandas().__repr__()
+        if preprocess.has_null_mask and not preprocess.dtype == "O":
+            output = (
+                preprocess.astype("O").fillna("null").to_pandas().__repr__()
+            )
         else:
-            output += "\nName: {}, dtype: {}".format(self.name, str_dtype)
-
-        if is_categorical_dtype(self.dtype):
-            cat = self.cat
-            str_dtype = cat.categories.dtype.name
-            categories = cat.categories.to_array()
-            separator = " < " if cat.ordered else ", "
-            desc = "({}, {})".format(len(categories), str_dtype)
-            categories = "[{}]".format(separator.join(categories))
-            output += "\nCategories {}: {}".format(desc, categories)
-
-        return output
-
-    def __str__(self):
-        return self.to_string(nrows=10)
-
-    def __repr__(self):
-        return "<cudf.Series nrows={} >".format(len(self))
+            output = preprocess.to_pandas().__repr__()
+        lines = output.split("\n")
+        if len(lines) > 1:
+            if lines[-1].startswith("N"):
```

Comment:
Can we make this safer?

Reply:
Do you mean to make the above conditionals function against things like `self.name` instead of just reflecting on the output string?

Reply:
I mean `startswith("N")` doesn't feel very safe and is prone to issues in the future, i.e.:
```
In [4]: pd.Series([1,2,3], index=pd.Index([11,12,13], name='Name'), name='name')
Out[4]:
Name
11    1
12    2
13    3
Name: name, dtype: int64
```

Reply:
Can we just make it `if lines[-1].startswith("Name: ")` to make it safer and clearer what's happening?

Reply:
```suggestion
            if lines[-1].startswith("Name: "):
```

Reply:
that's a good idear!